packages/panzer/disc-fe/src/evaluators/Panzer_GatherSolution_Epetra_impl.hpp

```diff
@@ -164,12 +166,13 @@ preEvaluate(typename TRAITS::PreEvalData d)
 
   // first try refactored ReadOnly container
   std::string post = useTimeDerivativeSolutionVector_ ? " - Xdot" : " - X";
-  if(d.gedc.containsDataObject(globalDataKey_+post)) {
-    ged = d.gedc.getDataObject(globalDataKey_+post);
-    RCP<EpetraVector_ReadOnly_GlobalEvaluationData> ro_ged = rcp_dynamic_cast<EpetraVector_ReadOnly_GlobalEvaluationData>(ged,true);
-
-    x_ = ro_ged->getGhostedVector_Epetra();
-
+  if (d.gedc.containsDataObject(globalDataKey_ + post))
+  {
+    ged = d.gedc.getDataObject(globalDataKey_ + post);
+    RCP<EpetraVector_ReadOnly_GlobalEvaluationData> ro_ged =
+      rcp_dynamic_cast<EpetraVector_ReadOnly_GlobalEvaluationData>(ged, true);
+    xOwned_   = ro_ged->getOwnedVector();
+    xGhosted_ = ro_ged->getGhostedVector_Epetra();
```

Comment:
If we're dealing with an `EpetraVector_ReadOnly_GlobalEvaluationData`, we want to set `xOwned_` and `xGhosted_` instead of `x_`.